docs/idle-crafting.rst

```diff
@@ -0,0 +1,56 @@
+idle-crafting
+=============
+
+.. dfhack-tool::
+    :summary: Allow dwarfs to automatically satisfy their need to craft objects.
+    :tags: fort gameplay
+
+This script allows dwarves to automatically satisfy their crafting needs. The
+script is configured through an overlay that is added to the main page of
+craftsdwarf's workshops, as described below.
+
+Usage
+-----
+
+::
+
+    idle-crafting status
+    idle-crafting disable
+    disable idle-crafting
+
+The ``status`` command prints statistics about the status of the tool and the
+satisfaction of "craft item" needs in your fort. Both variants of the
+``disable`` command disallow idle crafting at all workshops and disable the
+tool.
+
+Examples
+--------
+
+``idle-crafting -t 10000,1000,500 status``
+    Reset all thresholds to defaults and print status information.
+
+Options
+-------
+
+``-t <number list>``, ``--thresholds <number list>``
+    Sets the threshold(s) for the "craft item" need (i.e. the negated
+    ``focus_level``) at which the tool starts to generate crafting jobs for a
+    given unit. Units meeting earlier (higher) thresholds will be
+    prioritized. Defaults to ``10000,1000,500``.
+
+
+Overlay
+-------
+
+This script provides an overlay to the "Tasks" tab of craftsdwarf's workshops,
```

Comment:
We've discussed a little on Discord about placement. You didn't put this on the "Workers" tab because the tab is too crowded with `orders` overlay widgets, correct? Can we rearrange those widgets to make room for this one?

Another option is to hang the `idle-crafting` toggle off the side of the panel, like we do for the stocks screen:
![image](https://github.com/user-attachments/assets/b84e6be2-389a-468a-aec7-095d31b83f6e)

Reply:
While I do think that hanging toggles of to the side is acceptable for things that are essentially full-screen, I don't think that it's the right thing to do for building cards.

![craft_workers](https://github.com/user-attachments/assets/957ef0f3-ad4b-4a0d-b15e-c93c9aa149f5)

The only solution I can see for gaining a reasonable amount of space would be to split the "Permitted General Work Order Labors" into two columns. However, that is a list widget, so splitting this would complicate the code elsewhere.

Reply:
How much room does this widget need? Can it be made to fit in the area to the left of the Permitted General Work Order Labors panel?

Alternately, we could make the Permitted panel shorter and let the list be scrollable.

Reply:
I would say that the natural form is a single long line at the bottom. So, it technically does fit, but it does look a bit squished.
![squeeze](https://github.com/user-attachments/assets/4296e6fb-8c0c-457c-bc36-9e3a3f6eb7bf)

What about extending the "Permitted General Work Orders" to the full width, and moving their labels to the right of the list? This would provide the one line distance to make it not look squished, while also avoiding scrolling.

Reply:
I think we can shrink the labor list by 2 rows to accommodate this